engine_elasticsearch.go

```diff
@@ -90,30 +92,31 @@ func (ee *elasticEngine) Write(collection Collection, id string, cont Document)
 	}
 }
 
-func (ee *elasticEngine) Delete(collection Collection, id string) error {
+func (ee *elasticEngine) Delete(id string) (bool, error) {
 	if id == "" {
-		return errors.New("missing id")
+		return false, errors.New("missing id")
 	}
 
-	req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/%s/%s/%s", ee.baseURL, ee.indexName, collection.name, id), nil)
+	req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/%s/%s/%s", ee.baseURL, ee.indexName, ee.collectionName, id), nil)
 	if err != nil {
 		panic(err)
 	}
 	resp, err := ee.client.Do(req)
 	if err != nil {
 		fmt.Printf("e: %v\n", err)
-		return err
+		return false, err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != 400 && resp.StatusCode != 404 {
-		return fmt.Errorf("delete request failed with status %d", resp.Status)
+		return false, fmt.Errorf("delete request failed with status %d", resp.Status)
 	}
 
-	return nil
+	deleted := resp.StatusCode != 404
+	return deleted, nil
 }
 
-func (ee *elasticEngine) Count(collection Collection) int {
-	res, err := ee.client.Get(fmt.Sprintf("%s/%s/%s/_count", ee.baseURL, ee.indexName, collection.name))
+func (ee *elasticEngine) Count() (int, error) {
+	res, err := ee.client.Get(fmt.Sprintf("%s/%s/%s/_count", ee.baseURL, ee.indexName, ee.collectionName))
 	if err != nil {
 		panic(err)
```

Comment:
ES is still a work in progress TBH.  Anyway, I fixed most of them, but a handful remain for now.